r/inst/build_arrow_static.sh

```diff
@@ -45,6 +45,19 @@ else
   ARROW_DEFAULT_PARAM="OFF"
 fi
 
+if [ "$R_STATIC_DEPENDENCY_SOURCE" = "AUTO" ]; then
+  pkg-config --version >/dev/null 2>&1
+  if [ $? -eq 0 ]; then
+    ARROW_DEPENDENCY_SOURCE="AUTO"
+  else
+    ARROW_DEPENDENCY_SOURCE="BUNDLED"
+    echo "**** Warning: $R_STATIC_DEPENDENCY_SOURCE set to 'AUTO' but pkg-config not detected"
+    echo "**** Using bundled dependencies instead"
+  fi
```

Comment:
I don't think it makes sense to try and concoct a CI configuration that hits this else, but I'm curious if you've tried to do it locally and check that you get the expected behavior?

Reply:
I have not but happy to do so to test it. What do you think is the most straightforward approach?

Reply:
Should I just spin up a docker image?

Reply:
For this I would `ARROW_R_DEV=TRUE R_STATIC_DEPENDENCY_SOURCE=AUTO docker-compose build r` and then interactively enter the docker container that is built there, uninstall pkg-config and then run `/arrow/ci/scripts/r_test.sh` or try installing manually

Reply:
Oops, didn't see your second comment before I sent mine, yeah, exactly

Reply:
So atfer that, I ran this:
`ARROW_R_DEV=TRUE DEPENDENCY_SOURCE=AUTO docker run -it apache/arrow-dev:r-rhub-ubuntu-gcc-release-latest bash`

but couldn't find that script in the `arrow` directory.  Has something gone really wrong here?

Reply:
I think you want `ARROW_R_DEV=TRUE docker run -it apache/arrow-dev:r-rhub-ubuntu-gcc-release-latest -e ARROW_DEPENDENCY_SOURCE=AUTO /bin/bash` (get the right variable name, and pass it into the runtime environment with -e)

Reply:
oh you also need `-v $(pwd):/arrow` or something like that, that's what mounts the /arrow directory based on your local dir

Reply:
Thanks Neal!

Reply:
Just to clarify then, the reason why ARROW_DEPENDENCY_SOURCE needs to be passed in via `-e` but `ARROW_R_DEV` doesn't is:

- 'r' (defined in docker-compose.yml) is a service
- specifying environment variables ahead of calling `docker run` allows you to set env vars for running that service
- with `ARROW_R_DEV`, it doesn't matter whether you specify it via `-e` or via prefixing the call to `docker run` as it gets passed through anyway via the `environment` field of the `r` service
- with `ARROW_DEPENDENCY_SOURCE`,  it must be set via `-e` as there's nothing defined in the service to explicitly pass it through

Reply:
Now tested and working as expected when pkg-config isn't installed